canisters/src/vts/tests/agreement.rs

```diff
@@ -0,0 +1,53 @@
+use candid::{Decode, Encode, Principal};
+use ic_agent::Agent;
+use vts::VTSResult;
+
+use crate::agent::init_agent;
+
+mod agent;
+
+#[tokio::test]
+async fn test_create_agreement() {
+    let (agent, canister_id) = init_agent().await;
+
+    let name = "Test Agreement".to_string();
+    let vh_customer = Principal::anonymous();
+    let daily_usage_fee = "100".to_string();
+    let gas_price = "10".to_string();
+
+    let agreement_id_first =
+        create_agreement(&agent, &canister_id, &name, &vh_customer, &daily_usage_fee, &gas_price)
+            .await
+            .unwrap();
+    assert!(agreement_id_first > 0, "First agreement ID should be positive");
+
+    let result = 
+    create_agreement(&agent, &canister_id, &name, &vh_customer, &daily_usage_fee, &gas_price).await;
+    assert!(result.is_err(), "Should return error on duplicate agreement");
```

Comment:
@lavrd maybe you have an idea why I get this error?

```
thread 'test_create_agreement' panicked at canisters/src/vts/tests/agreement.rs:21:14:
called `Result::unwrap()` on an `Err` value: AlreadyExists
```

Reply:
At the moment we don't have auto cleaner for our canister. So maybe you need to restart your node and deploy canister from scratch?